src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -3271,7 +3271,7 @@ void CodeGen::genCall(GenTreeCall* call)
 
     // If there is nothing next, that means the result is thrown away, so this value is not live.
     // However, for minopts or debuggable code, we keep it live to support managed return value debugging.
-    if ((call->gtNext == nullptr) && !compiler->opts.MinOpts() && !compiler->opts.compDbgCode)
+    if ((call->gtNext == nullptr) && !compiler->opts.OptimizationDisabled() && !compiler->opts.compDbgCode)
```

Comment:
> `!compiler->opts.OptimizationDisabled()`

Nit: the double negation is hard to read. Can these (there is a number instances) be switched to `OptimizationEnabled`?

Reply:
Right, but I didn't want to distract code reviewers with that. I think we'd better do it separately, for now it's easier to review because I simply replaced MinOpts with the same OptimizationDisabled so no need to validate it 🙂